CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 # CMake setup for Unicorn 2.
 # By Huitao Chen & Nguyen Anh Quynh, 2019-2020
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
This breaks some slightly older Ubuntu.

Reply:
https://stackoverflow.com/a/49921543 looks like "slightly" is pretty old...

Reply:
fwiw i raised it because it allows using `CMAKE_MSVC_RUNTIME_LIBRARY`, and changes behavior around how cmake handles the msvcrt settings. On older cmake, they supply a default value which is hard/impossible to override, and it is something that should be done for rust.

This also relates to adding the CARGO variable, which is mostly used for configuring msvcrt setting. I can see if I can move it somewhat back into build.rs, but it will still require more recent cmake version either way.

Reply:
> https://stackoverflow.com/a/49921543 looks like "slightly" is pretty old...

Ubuntu 18 seems not that old, at least it's not EOL.

Reply:
> fwiw i raised it because it allows using `CMAKE_MSVC_RUNTIME_LIBRARY`, and changes behavior around how cmake handles the msvcrt settings. On older cmake, they supply a default value which is hard/impossible to override, and it is something that should be done for rust.
> 
> This also relates to adding the CARGO variable, which is mostly used for configuring msvcrt setting. I can see if I can move it somewhat back into build.rs, but it will still require more recent cmake version either way.

Is there any reference for this? Let me also check how could we work out this.

Reply:
> > https://stackoverflow.com/a/49921543 looks like "slightly" is pretty old...
> 
> Ubuntu 18 seems not that old, at least it's not EOL.

i've changed it back to min-version = 3.1, but if compiling for msvc it will either enable the new behavior or error out if < 3.15